webv2/api/schema.go

```
@@ -1028,11 +1027,15 @@ func parentTableHelper(tableId string, update bool) *types.TableInterleaveStatus
 
 			colIdNotInOrder := checkPrimaryKeyOrder(tableId, refTableId, fk)
 
-			if update && sp.ParentId == "" && colIdNotInOrder == "" {
+			if update && sp.ParentTable.Id == "" && colIdNotInOrder == "" {
 				usedNames := sessionState.Conv.UsedNames
 				delete(usedNames, strings.ToLower(sp.ForeignKeys[i].Name))
-				sp.ParentId = refTableId
-				sp.ForeignKeys = utilities.RemoveFk(sp.ForeignKeys, sp.ForeignKeys[i].Id)
+				sp.ParentTable.Id = refTableId
+				sp.ParentTable.OnDelete = onDelete
+				sp.ForeignKeys, err = utilities.RemoveFk(sp.ForeignKeys, sp.ForeignKeys[i].Id, sessionState.Conv.SrcSchema[tableId], tableId)
+				if err != nil {
+					continue
```

Do we want to silently ignore this error?